src/zenml/prompts/prompt_comparison.py

```diff
@@ -0,0 +1,215 @@
+#  Copyright (c) ZenML GmbH 2024. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Simple prompt comparison utilities for A/B testing."""
+
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from pydantic import BaseModel
+
+from zenml.logger import get_logger
+from zenml.prompts import Prompt
+
+logger = get_logger(__name__)
+
+
+class PromptComparisonResult(BaseModel):
+    """Result of comparing two prompts."""
+
+    prompt_a_version: str
+    prompt_b_version: str
+    prompt_a_metrics: Dict[str, Any]
+    prompt_b_metrics: Dict[str, Any]
+    winner: Optional[str] = None
+    comparison_summary: Dict[str, Any] = {}
+
+
+def compare_prompts(
```

Comment:
this function / behaviour needs a LOT more documentation. 

actually I maybe would even consider this out of scope. I think it's more complicated than this function would suggest.